byway/src/main/webapp/js/recommendations.js

```diff
@@ -29,32 +29,29 @@ const MAX_RECOMMENDATIONS = 1;
 // Holds regions google.maps.LatLng objects
 let regions = [];
 
+/** Uses the route from the directionsService request to find suitable recommendations
+ *  along each leg 
+ */
+async function getRecommendations(){
+  try{
+    let result = await getRouteForTrip();
+    findRegions(result);
+    loadRecommendations();
+  } catch (error) {
+      console.error(error);
+  }
+}
+
 /**
  * Creates round-trip route with waypoints that loads onto the map.
  * Partitions the route into regions, used to load recommendations
  * around. Orders the waypoints for efficiency and updates trip logistics.
  */
 function calcMainRoute() {
   resetUserAlerts();
-  const request = {
-    origin:  start.name,
-    destination: end.name,
-    travelMode: 'DRIVING',
-    waypoints:  route.map(waypoint => ({location: waypoint.geometry.location})),
-    optimizeWaypoints: true
-  };
   addMainStopsToRegions();
-  directionsService.route(request, function(result, status) {
-    if (status == 'OK') {
-      directionsRenderer.setDirections(result);
-      findRegions(result);
-      loadRecommendations();
-      orderWaypoints(result);
-      updateDistanceTime(result);
-    } else {
-      alert("Could not calculate route due to: " + status);
-    }
-  });
+  getRecommendations();
+  updatePageInfo();
```

Comment:
updatePageInfo calls getRouteForTrip, but so does getRecommendations, so it makes two uses of the directions API. Even though this double-call seems to happen once initially, since we getRecs once, might be worth looking into.

Reply:
Yeah... I noticed this as well but I wasn't sure how to resolve it since they both need to use the directionsServices but are called at different times. I might wait on future suggestions from Zach and Landon.